src/drivers/distance_sensor/lightware_sf45_serial/lightware_sf45_serial.cpp

```diff
@@ -207,11 +245,17 @@ int SF45LaserSerial::collect()
 
 			ret = ::read(_fd, &readbuf[0], 10);
 
+			if (ret < 0) {
+				PX4_ERR("ERROR (unknown sensor data): %d", ret);
+				perf_count(_comms_errors);
+				perf_end(_sample_perf);
+				return ret;
+			}
 		}
 	}
 
 	if (ret < 0) {
```

Comment:
Question: Can this path even be reached? It seems like whenever you get a `ret < 0` you directly `return ret`.